objectstore-service/src/lib.rs

```diff
@@ -223,18 +250,35 @@ impl StorageService {
     pub async fn delete_object(&self, path: &ObjectPath) -> anyhow::Result<()> {
         let start = Instant::now();
 
-        let res1 = self.0.high_volume_backend.delete_object(path).await;
-        let res2 = self.0.long_term_backend.delete_object(path).await;
+        if let Some((metadata, _stream)) = self.0.high_volume_backend.get_object(path).await? {
+            if metadata.is_redirect_tombstone == Some(true) {
+                self.0.long_term_backend.delete_object(path).await?;
```

Comment:
Let's add a comment that we accept the risk of orphans in case this delete fails. I can see this challenged in the future, and it's worth noting that we considered this case.